frontend/src/components/table.tsx

```diff
@@ -0,0 +1,137 @@
+import React, { useState } from 'react';
+import '../styles/table.css';
+
+interface driversSettings {
+  name: string;
+  netid: string;
+  email: string;
+  phone: string;
+}
+
+interface FormProps {
+  onClick: ((newDriver: driversSettings) => void);
+}
+
+function deleteEntry(netid: string, driverList: driversSettings[]) {
+  const newDriverList = [];
+  for (let i = 0; i < driverList.length; i += 1) {
+    if (driverList[i].netid !== netid) {
+      newDriverList.push(driverList[i]);
+    }
+  }
+  return newDriverList;
+}
+
+function renderTableHeader() {
+  return (
+    <tr>
+      <th>Name</th>
+      <th>Netid</th>
+      <th>Email</th>
+      <th>Phone Number</th>
+    </tr>
+  );
+}
+
+function addDriver(newDriver: driversSettings, allDrivers: driversSettings[]) {
+  const updatedDrivers = [...allDrivers, newDriver];
+  return updatedDrivers;
+}
+
+const Form = (props: FormProps) => {
+  const $newDriver = { name: '', netid: '', email: '', phone: '' };
+
+  const handleInput = (evt: any) => {
+    evt.preventDefault();
+    const $fieldName = evt.target.name;
+    const $fieldValue = evt.target.value;
+    if ($fieldName === 'name') { $newDriver.name = $fieldValue; } else if ($fieldName === 'netid') { $newDriver.netid = $fieldValue; } else if ($fieldName === 'email') { $newDriver.email = $fieldValue; } else { $newDriver.phone = $fieldValue; }
```

Comment:
Do not have this all on one line. Each line shouldn't really be longer than 80 characters. I recommend putting each if, else if, or else on a new line.